### PR TITLE
[Communication - Job] Migrate to Test proxy

### PR DESCRIPTION
**Pull request by samvaity**
- Update for upsert operations to expect status codes {200, 201} - confirmed by @cparisineti (To be updated in swagger, follow up issue - https://github.com/Azure/azure-sdk-for-java/issues/35706)
- Disabled unassignJob test - Follow up issue - https://github.com/Azure/azure-sdk-for-java/issues/35706

### CODE REVIEW

**Comment by cparisineti** [comment]
Test proxy seems to be failing the tests. Is this a mandatory migration? Please link me a doc related to this in Java. I can look into what's causing the test failures. Thanks.

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/JobRouterAdministrationsImpl.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ public final class JobRouterAdministrationsImpl {
     @ServiceInterface(name = "AzureCommunicationSe")
     public interface JobRouterAdministrationsService {
         @Patch("/routing/classificationPolicies/{id}")
-        @ExpectedResponses({200})
+        @ExpectedResponses({200,201})
```
Comment (samvaity) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/JobRouterAdministrationsImpl.java:72]: @cparisineti Is this expected?

**Comment by samvaity** [comment]
> Test proxy seems to be failing the tests. Is this a mandatory migration? Please link me a doc related to this in Java. I can look into what's causing the test failures. Thanks.

Test Proxy is just a testing framework update. It should not be interfering with the actual test cases.
Mono tracking issue to update to Test Proxy for all services - https://github.com/Azure/azure-sdk-for-java/issues/33342

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/JobRouterAdministrationsImpl.java (hunk 1 above)
Comment (cparisineti) [reply sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/JobRouterAdministrationsImpl.java:72]: I see both 200, 201 in the service code  - https://skype.visualstudio.com/SCC/_git/ic3_spool_ccaap?path=/src/Router.Api/V2022_07_18_preview/Controllers/ClassificationPoliciesController.cs 

but the swagger indeed only shows only 200 - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/JobRouter/preview/2022-07-18-preview/communicationservicejobrouter.json

**Review by cparisineti — APPROVED** [review]

**Pull request merged**